vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -367,6 +367,8 @@ import { isFocusable } from '@vaadin/grid/src/vaadin-grid-active-item-mixin.js';
             // workaround: sometimes grid-element gives page index that overflows
             page = Math.min(page, Math.floor(cache[parentUniqueKey].size / grid.pageSize));
 
+            // Ensure grid isn't in loading state when the callback executes
+            ensureSubCacheQueue = [];
```

Comment:
I see the commit message and the change, but I cannot make the connection between them. Could you explain it shortly?

Reply:
Sorry, should've added a better explanation for the fix.

There was a [failing test case](https://teamcity.vaadin.com/viewLog.html?buildId=112823&buildTypeId=VaadinFlowComponents_PullRequestValidation) related to recalculating column auto widths. The item hierarchy was preloaded and rendered synchronously (as desired).There's a [check](https://github.com/vaadin/web-components/blob/4c388aeec6623869c70896c909799804fc95d0f9/packages/grid/src/vaadin-grid.js#L538) in `<vaadin-grid>` that prevents running `_recalculateColumnWidths` in case the cache is in loading state. The `gridConnector` [overrides](https://github.com/vaadin/flow-components/blob/8cf147695dbe55a1b4589b62013c14d0ee96d496/vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js#L39-L43) `isLoading` to also check the `ensureSubCacheQueue` and in this case it wasn't empty so `_recalculateColumnWidths` would not get invoked.

To fix this, we're emptying the queue here before invoking the callback to make sure `_recalculateColumnWidths` is not blocked by the `ensureSubCacheQueue` array (which gets emptied again right after the callback).

Reply:
Thanks, it's clear now.